spec.bs

```diff
@@ -13,3 +13,538 @@ Group: patcg
+      1. For each |topic| in |topics|:
+          1. Let |topicItem| be a Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-integers">Integer</a> with value |topic|["{{BrowsingTopic/topic}}"].
+          1. Let |topicParameters| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-parameters">Parameters</a>.
+          1. Set |topicParameters|["<code>version</code>"] to |topic|["{{BrowsingTopic/version}}"].
+          1. Set |topicParameters|["<code>config_version</code>"] to |topic|["{{BrowsingTopic/configVersion}}"].
+          1. Set |topicParameters|["<code>model_version</code>"] to |topic|["{{BrowsingTopic/modelVersion}}"].
+          1. Set |topicParameters|["<code>taxonomy_version</code>"] to |topic|["{{BrowsingTopic/taxonomyVersion}}"].
+          1. Associate |topicParameters| with |topicItem|.
+          1. Insert |topicItem| to |headerStructuredFields|.
+      1. Run the [=set a structured field value=] algorithm with tuple field <var ignore=''>name</var> set to \`<code>Sec-Browsing-Topics</code>\`, <var ignore=''>structuredValue</var> set to |headerStructuredFields|, and <var ignore=''>list</var> set to |request|’s [=header list=].
+
+    Note: In Chrome's experimentation phase, it will additionally require a valid <a href="https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/explainer.md">Origin Trial</a> token to exist in |initiatorWindow|'s <a data-cite="!HTML#concept-document-window">associated document</a> for the request to be eligible for topics.
+  </div>
+
+  <h3 id="the-observe-browsing-topics-http-response-header-header">The \`<code>Observe-Browsing-Topics</code>\` HTTP response header</h3>
+
+  The \`<dfn export http-header><code>Observe-Browsing-Topics</code></dfn>\` HTTP response header can be used to record a caller's topics observation.
+
+  <div algorithm>
+    To <dfn>handle topics response</dfn>, given a [=response=] |response| and a [=request=] request:
+
+    1. If |request|'s [=request/header list=] does not [=list/contain=] [:Sec-Browsing-Topics:] (implying the |request|'s [=request/current URL=] is not eligible for topics), then return.
+    1. Let |topLevelDocument| be |request|'s [=request/window=]'s [=environment settings object/global object=]'s [=Window/navigable=]'s [=navigable/top-level traversable=]'s [=navigable/active document=].
+    1. Let |callerOrigin| be |request|'s [=request/current URL=]'s [=url/origin=].
+    1. Let |list| be |response|'s [=response/header list=].
+    1. Let |values| be the result of [=getting, decoding, and splitting=] [:Observe-Browsing-Topics:] from |list|.
+    1. Let |observe| be the result of <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-parsing-a-boolean">parsing</a> |values| into Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-booleans">Boolean</a>.
```

Comment:
Parsing takes a string. I think you need to not get/decode/split, so that it stays a string instead of becoming a list of strings.